lib/streamlit/server/server.py

```diff
@@ -210,33 +214,28 @@ def start_listening_tcp_socket(http_server):
 
     if call_count >= MAX_PORT_SEARCH_RETRIES:
         raise RetriesExceeded(
-            "Cannot start Streamlit server. Port %s is already in use, and "
-            "Streamlit was unable to find a free port after %s attempts.",
-            port,
-            MAX_PORT_SEARCH_RETRIES,
+            f"Cannot start Streamlit server. Port {port} is already in use, and "
+            f"Streamlit was unable to find a free port after {MAX_PORT_SEARCH_RETRIES} attempts.",
         )
 
 
-class Server(object):
-
-    _singleton = None  # type: Optional[Server]
+class Server:
+    _singleton: Optional["Server"] = None
 
     @classmethod
-    def get_current(cls):
+    def get_current(cls) -> "Server":
```

Comment:
is the parameter needed?

Reply:
Yes, it's accessing the `cls._singleton` variable

Reply:
But you've replaced it with `Server._singleton`